File: 프로그래머스/연습문제/Lv1/문자열 나누기/문자열 나누기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string s) {
    int cnt=0, idx=0;
    char x;
    for(int i=0; i<s.length(); i++){
        x = s[i];
        cnt++;
        int xi=1, yi=0;
        for(idx=i+1; idx<s.length() && xi!=yi; idx++){
            if(s[idx] == x) xi++;
            else yi++;
        }
        
        i = idx-1;
    }
    
    return cnt;
}

File: 프로그래머스/연습문제/Lv1/문자열 나누기/문자열 나누기.py
def solution(s):
    cnt,idx = 0,0
    while idx < len(s):
        x = s[idx]
        cnt += 1
        xi,yi = 1,0

        idx += 1
        while idx<len(s) and xi!=yi:
            if s[idx] == x: xi += 1
            else: yi += 1
            idx += 1

    return cnt
